### PR TITLE
[WIP - DO NOT MERGE] Merge policies

### PR DESCRIPTION
**Pull request by saiyan86**
**What this PR does / why we need it**:

This PR merges all networkpolicies targeting the same podselector together and simplifies logic of parsing code.

**Special notes for your reviewer**:

DO NOT MERGE

**Release note**:

```release-note
N/A
```

### CODE REVIEW

**Comment by saiyan86** [comment]
@kalebmorris

File: npm/translatePolicy.go [hunk 1]
```diff
@@ -0,0 +1,800 @@
...
+					util.IptablesCommentFlag,
+					"ALLOW-ALL-FROM-" + protPortPair.port + "-PORT-OF-" +
+					targetSelectorComment +
+					"-TO-JUMP-TO-" + util.IptablesAzureEgressToChain,
+				)
+				entries = append(entries, entry)
+			}
+		} else {
+			entry := &iptm.IptEntry{
+				Chain: util.IptablesAzureEgressToChain,
```
Comment (kalebmorris) [inline npm/translatePolicy.go]: I think this is supposed to be util.IptablesAzureEgressPortChain

File: npm/plugin/main.go [hunk 2]
```diff
@@ -50,22 +52,23 @@ func main() {
...
 
 	factory := informers.NewSharedInformerFactory(clientset, time.Hour*24)
 
 	npMgr := npm.NewNetworkPolicyManager(clientset, factory, version)
+
+	go npMgr.SendNpmTelemetry()
+
+	time.Sleep(time.Second * waitForTelemetryInSeconds)
+
 	err = npMgr.Start(wait.NeverStop)
```
Comment (jaer-tsun) [inline npm/plugin/main.go]: keep err = ... in the same line as if since it's just 1 return

File: npm/pod.go [hunk 3]
```diff
@@ -136,24 +129,18 @@ func (npMgr *NetworkPolicyManager) DeletePod(podObj *corev1.Pod) error {
...
 	podIP := podObj.Status.PodIP
-	log.Printf("POD DELETING: %s/%s/%s\n", podNs, podName, podNodeName)
+	log.Printf("POD DELETING: [%s/%s/%s%+v%s]", podNs, podName, podNodeName, podLabels, podIP)
 
 	// Delete pod from ipset
 	ipsMgr := npMgr.nsMap[util.KubeAllNamespacesFlag].ipsMgr
 	// Delete the pod from its namespace's ipset.
-	if err = ipsMgr.DeleteFromSet(podNs, podIP); err != nil {
-		log.Printf("Error deleting pod from namespace ipset.\n")
+	if err = ipsMgr.DeleteFromSet("ns-" + podNs, podIP); err != nil {
```
Comment (jaer-tsun) [inline npm/pod.go]: remove "ns-"

File: npm/translatePolicy.go [hunk 4]
```diff
@@ -0,0 +1,1035 @@
...
+	protocol string
+	port     string
+}
+
+func craftPartialIptEntrySpecFromPort(portRule networkingv1.NetworkPolicyPort, sPortOrDPortFlag string) []string {
+	partialSpec := []string{}
+	if portRule.Protocol != nil {
+		partialSpec = append(
+			partialSpec,
+			util.IptablesProtFlag,
```
Comment (jaer-tsun) [inline npm/translatePolicy.go]: Port* instead of Prot

File: npm/translatePolicy.go [hunk 5]
```diff
@@ -0,0 +1,1035 @@
...
+			string(*portRule.Protocol),
+		)
+	}
+
+	if portRule.Port != nil {
+		partialSpec = append(
+			partialSpec,
+			sPortOrDPortFlag,
```
Comment (jaer-tsun) [inline npm/translatePolicy.go]: since it's src or dest, can this just be portFlag?

File: npm/translatePolicy.go [hunk 6]
```diff
@@ -0,0 +1,1035 @@
...
+			portRule.Port.String(),
+		)
+	}
+
+	return partialSpec
+}
+
+func craftPartialIptablesCommentFromPort(portRule networkingv1.NetworkPolicyPort, sPortOrDPortFlag string) string {
```
Comment (jaer-tsun) [inline npm/translatePolicy.go]: sPortOrDPortFlag isn't used here

File: npm/translatePolicy.go [hunk 7]
```diff
@@ -0,0 +1,1035 @@
...
+					srcOrDstFlag,
+				}
+			}
+
+			return spec
+		}
+	}
+
+	for i, _ := range ops {
+		spec = append(spec, craftPartialIptEntrySpecFromOpAndLabel(ops[i], labels[i], srcOrDstFlag, isNamespaceSelector)...)
```
Comment (jaer-tsun) [inline npm/translatePolicy.go]: is ops and labels guaranteed to be same length?

File: npm/translatePolicy.go [hunk 8]
```diff
@@ -0,0 +1,1035 @@
...
+			if len(rule.From) == 0 {
+				fromRuleExists = true
+				allowExternal = true
+			}
+
+			for _, fromRule := range rule.From {
+				if fromRule.PodSelector != nil ||
+					fromRule.NamespaceSelector != nil ||
+					fromRule.IPBlock != nil {
+					fromRuleExists = true
```
Comment (jaer-tsun) [inline npm/translatePolicy.go]: break after setting to true

File: npm/translatePolicy.go (hunk 6 above)
Comment (saiyan86) [reply npm/translatePolicy.go]: Let's keep it this way for now. There could be upstream specification changes.

File: npm/translatePolicy.go (hunk 7 above)
Comment (saiyan86) [reply npm/translatePolicy.go]: yes

File: npm/translatePolicy.go (hunk 8 above)
Comment (saiyan86) [reply npm/translatePolicy.go]: done

**Pull request merged**